src/plugins/discover/public/application/components/doc/use_es_doc_search.ts

```diff
@@ -42,6 +42,7 @@ export function buildSearchBody(
     fields: useNewFieldsApi ? [{ field: '*', include_unmapped: 'true' }] : undefined,
     script_fields: computedFields.scriptFields,
     docvalue_fields: computedFields.docvalueFields,
+    runtime_mappings: useNewFieldsApi && runtimeFields ? runtimeFields : {},
```

Comment:
Is this related to the add/delete change? It seems like this is an unrelated bug fix, right?

Reply:
No, this is needed for a single doc view, so that runtime index pattern fields are shown in a single doc view

Reply:
But this is not related to adding/deleting runtime fields from within Discover, right? if it's a bug which would be in there without the changes of this PR (and especially if it's in 7.12) I think it's worth at least a separate issue closed by this PR and a fix release note so we can point to something.